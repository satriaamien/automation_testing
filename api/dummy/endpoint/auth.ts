import dotenv from "dotenv";
dotenv.config();
import axios from "axios";
const BASE_URL = process.env.REQRES_URL; // Mengambil nilai dari .env

export const page = (page: number) =>
  axios.get(`${BASE_URL}api/users?page=${page}`);
