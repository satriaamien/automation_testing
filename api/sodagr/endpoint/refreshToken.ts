import supertest from "supertest";
import dotenv from "dotenv";
import {
  RefreshToken,
  RefreshTokenBooleanType,
  RefreshTokenNumberType,
} from "../interfaces/refreshToken";
dotenv.config();
const sodagr = process.env.SODAGR;
const url = supertest(sodagr!);

export const refreshToken = (payload: RefreshToken) =>
  url
    .post("/api/v1/refresh-token")
    .set("Content-Type", "application/json")
    .send(payload);

export const refreshTokenBoolean = (payload: RefreshTokenBooleanType) =>
  url
    .post("/api/v1/refresh-token")
    .set("Content-Type", "application/json")
    .send(payload)
    .expect(400);

export const refreshTokenNumber = (payload: RefreshTokenNumberType) =>
  url
    .post("/api/v1/refresh-token")
    .set("Content-Type", "application/json")
    .send(payload)
    .expect(400);
