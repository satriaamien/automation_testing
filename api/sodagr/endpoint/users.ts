import supertest from "supertest";
import dotenv from "dotenv";
dotenv.config();
const sodagr = process.env.SODAGR;
const url = supertest(sodagr!);

export const getAllUsers = (token: string) =>
  url
    .get("/api/v1/users")
    .set("Content-Type", "application/json")
    .set("Authorization", `Bearer ${token}`);
