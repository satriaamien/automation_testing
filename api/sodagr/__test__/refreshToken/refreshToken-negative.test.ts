import randomstring from "randomstring";
import {
  refreshToken,
  refreshTokenBoolean,
  refreshTokenNumber,
} from "../../endpoint/refreshToken";
import { login } from "../../endpoint/login_test";
const username = process.env.usernameSodagr;
const password = process.env.passwordSodagr;

describe("user failed to refresh token", () => {
  test("refresh token with invalid type boolean", async () => {
    const payloadRefresher = {
      refreshToken: true,
    };
    const response = await refreshTokenBoolean(payloadRefresher);
    expect(response.body).toMatchSnapshot();
  });
  test("refresh token with invalid type number", async () => {
    const payloadRefresher = {
      refreshToken: 123456789,
    };
    const response = await refreshTokenNumber(payloadRefresher);
    expect(response.body).toMatchSnapshot();
  });
  test("refresh token with empty payload", async () => {
    const payloadRefresher = {
      refreshToken: "",
    };
    const response = await refreshToken(payloadRefresher);
    expect(response.status).toBe(401);
    expect(response.body).toMatchSnapshot();
  });
  test("refresh token with expired token", async () => {
    const payloadRefresher = {
      refreshToken:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjU2LCJlbWFpbCI6Imx1cW1hbnVsQGViY29ubmVjdC5jb20iLCJuYW1lIjoiTHVxbWFudWwgSGFraWVtIiwiaWF0IjoxNzQ2NTM1MjEyLCJleHAiOjE3NDY1Mzg4MTJ9.1n8C-WttBY2F5EmFQFDrVo86pDMl0BdLVyqkjaW0LO4",
    };
    const response = await refreshToken(payloadRefresher);
    expect(response.body).toMatchSnapshot();
  });
  test("refresh token with random string", async () => {
    const randomString = randomstring.generate();
    const payloadRefresher = {
      refreshToken: randomString,
    };
    const response = await refreshToken(payloadRefresher);
    expect(response.status).toBe(401);
    expect(response.body).toMatchSnapshot();
  });
});
