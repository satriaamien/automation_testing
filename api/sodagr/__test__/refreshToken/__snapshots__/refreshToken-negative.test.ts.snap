// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user failed to refresh token refresh token with empty payload 1`] = `
{
  "message": "Refresh token is invalid or has expired",
  "success": false,
}
`;

exports[`user failed to refresh token refresh token with expired token 1`] = `
{
  "message": "Refresh token is invalid or has expired",
  "success": false,
}
`;

exports[`user failed to refresh token refresh token with invalid type boolean 1`] = `
{
  "message": "refreshToken: refreshToken must be a string",
  "success": false,
}
`;

exports[`user failed to refresh token refresh token with invalid type number 1`] = `
{
  "message": "refreshToken: refreshToken must be a string",
  "success": false,
}
`;

exports[`user failed to refresh token refresh token with random string 1`] = `
{
  "message": "Refresh token is invalid or has expired",
  "success": false,
}
`;
